src/www/carp_status.php

```diff
@@ -163,26 +163,28 @@ interface_vip_bring_down($vip);
                 </tr>
 <?php
                 else:
+                  $interfaces_details = legacy_interfaces_details();
                   foreach ($a_vip as $carp):
                     if ($carp['mode'] != "carp") {
                         continue;
                     }
                     $icon = '';
-                    $intf_status = get_carp_interface_status("{$carp['interface']}_vip{$carp['vhid']}");
+                    $intf_carp_statuses = $interfaces_details[get_real_interface($carp['interface'])]['carp'];
+                    $intf_status = $intf_carp_statuses[array_search($carp['vhid'], array_column($intf_carp_statuses, 'vhid'))]['status'];
                     if (($carpcount > 0 && !$status)) {
                         $icon = "fa fa-remove fa-fw text-danger";
-                        $intf_status = gettext('DISABLED');
-                    } elseif ($intf_status == gettext('MASTER')) {
+                        $intf_status = 'DISABLED';
+                    } elseif ($intf_status == 'MASTER') {
                         $icon = "fa fa-play fa-fw text-success";
-                    } elseif ($intf_status == gettext('BACKUP')) {
+                    } elseif ($intf_status == 'BACKUP') {
                         $icon = "fa fa-play fa-fw text-muted";
-                    } elseif ($intf_status == gettext('INIT')) {
+                    } elseif ($intf_status == 'INIT') {
                         $icon = "fa fa-info-circle fa-fw";
                     }?>
                 <tr>
                   <td><?=convert_friendly_interface_to_friendly_descr($carp['interface']) . "@{$carp['vhid']}" ;?></td>
                   <td><?=$carp['subnet'];?></td>
-                  <td><span class="<?=$icon;?>"></span> <?=$intf_status;?></td>
+                  <td><span class="<?=$icon;?>"></span> <?=gettext($intf_status);?></td>
```

Comment:
this will be broken, no static text in gettext(), part of why the weirdness existed before

Reply:
(I don't mind dropping the gettext altogether)

Reply:
> this will be broken, no static text in gettext(), part of why the weirdness existed before

Do you mean `gettext()` doesn't accept dynamic text, only static one? I haven't caught any php warnings/errors on my test system with gettext in place.

Reply:
PHP doesn't care, but translation generator reads all static text from gettext() calls to prepare a translation. You can't read variables from there since the text is never executed/analysed.

Reply:
Got it, thank you. Added a few commits to resolve these translation issues.